centreon/lib/Centreon/Object/Host/Host.php

```diff
@@ -45,4 +46,245 @@ class Centreon_Object_Host extends Centreon_Object
     protected $table = "host";
     protected $primaryKey = "host_id";
     protected $uniqueLabelField = "host_name";
+
+    /**
+     * Update host table
+     *
+     * @param $hostId
+     * @param array $params
+     * @return void
+     */
+    public function update($hostId, $params = [])
+    {
+        parent::update($hostId, $params);
+        $vaultConfiguration = $this->getVaultConfiguration();
+        if ($vaultConfiguration !== null) {
+            try {
+                $httpClient = new CentreonRestHttp();
+                $centreonLog = new CentreonUserLog(-1, $this->db);
+                $clientToken = $this->authenticateToVault($vaultConfiguration, $centreonLog, $httpClient);
+                $hostSecrets = $this->getHostSecretsFromVault(
+                    $vaultConfiguration,
+                    $hostId,
+                    $clientToken,
+                    $centreonLog,
+                    $httpClient
+                );
+
+                if (
+                    array_key_exists('host_snmp_community', $params)
```

Comment:
This condition is the same in the `elseif`.
Why not using it **before** the http calls to avoid them when possible ?